spec/defines/router_log_spec.rb

```diff
@@ -1,58 +1,97 @@
+      it do
+        verify_concat_fragment_exact_contents(catalogue, 'qdrouter+log_example.conf', [
+          'log {',
+          '    module: DEFAULT',
+          '    enable: info+',
+          '    timestamp: true',
+          '    output: /var/log/qdrouterd.log',
+          '}',
+        ])
+      end
     end
-  end
 
-  context 'with dependencies' do
-    let :facts do
-      on_supported_os['redhat-7-x86_64']
-    end
+    context 'with logging to file' do
+      let :params do
+        super().merge(
+          module: "DEFAULT",
+          level: "debug+",
+          timestamp: false,
+          output: "/var/log/qpid.log",
+        )
+      end
 
-    let :pre_condition do
-      'include qpid::router'
+      it 'should have log fragment' do
+        verify_concat_fragment_exact_contents(catalogue, 'qdrouter+log_example.conf', [
+          'log {',
+          '    module: DEFAULT',
+          '    enable: debug+',
+          '    timestamp: false',
+          '    output: /var/log/qpid.log',
+          '}'
+        ])
+      end
     end
 
-    it { is_expected.to compile.with_all_deps }
-
-    it do
-      verify_concat_fragment_exact_contents(catalogue, 'qdrouter+log_example.conf', [
-        'log {',
-        '    module: DEFAULT',
-        '    enable: info+',
-        '    timestamp: true',
-        '    output: /var/log/qdrouterd.log',
-        '}',
-      ])
+    context 'with logging to syslog' do
+      let :params do
+        super().merge(
+          module: "DEFAULT",
+          level: "debug+",
+          timestamp: false,
+          output: "syslog",
+        )
+      end
+
+      it 'should have log fragment' do
+        verify_concat_fragment_exact_contents(catalogue, 'qdrouter+log_example.conf', [
+          'log {',
+          '    module: DEFAULT',
+          '    enable: debug+',
+          '    timestamp: false',
+          '    output: syslog',
+          '}'
+        ])
+      end
     end
+  end
+
+  context 'with dependencies' do
+    on_os_under_test.each do |os, facts|
+      context "on #{os}" do
+        let(:pre_condition) { 'include qpid::router' }
+
+        it { is_expected.to compile.with_all_deps }
 
-    it { is_expected.to contain_concat_fragment('qdrouter+log_example.conf') }
+        it do
+          verify_concat_fragment_exact_contents(catalogue, 'qdrouter+log_example.conf', [
+            'log {',
+            '    module: DEFAULT',
+            '    enable: info+',
+            '    timestamp: true',
+            '    output: /var/log/qdrouterd.log',
+            '}',
+          ])
+        end
 
-    it do
-      is_expected.to contain_concat('/etc/qpid-dispatch/qdrouterd.conf')
-        .that_notifies('Service[qdrouterd]')
-        .that_subscribes_to('Concat_fragment[qdrouter+log_example.conf]')
+        it do
+          is_expected.to contain_concat_fragment('qdrouter+log_example.conf')
+            .that_notifies(['Concat[/etc/qpid-dispatch/qdrouterd.conf', 'Service[qdrouterd]'])
```

Comment:
this should be `.that_notifies(['Concat[/etc/qpid-dispatch/qdrouterd.conf]', 'Service[qdrouterd]'])`